lib/compiler/src/engine/unwind/systemv.rs

```diff
@@ -141,6 +157,20 @@ impl UnwindRegistry {
     unsafe fn register_frames(&mut self, eh_frame: &[u8]) {
         #[cfg(all(target_os = "macos", target_arch = "aarch64"))]
         {
+            // Acquire the unregistry lock to avoid interleaved registrations/deregistrations.
+            let mut unregistry = GLOBAL_UNREGISTRY.lock().unwrap();
```

Comment:
Here we would iterate the `PublishedUnwindRegistry`, and call `PublishedUnwindRegistry::unregister`?